### PR TITLE
Enable linking of FB account with pre-created account from mobile

### PR DESCRIPTION
**Pull request by elropero**

### CODE REVIEW

File: middleware/sh-passport-configurator.js [hunk 1]
```diff
@@ -0,0 +1,584 @@
...
+          if (link) {
+            if (req.user) {
+              self.userCredentialModel.link(
+                req.user.id, provider, authScheme, profile,
+                {
+                  accessToken: accessToken,
+                  refreshToken: refreshToken,
+                }, options, done);
+            } else {
+              // NOTE: req.user ends up being undefined here, even if the client
```
Comment (elropero) [inline middleware/sh-passport-configurator.js]: This is a key change I needed to make.

File: middleware/sh-passport-configurator.js [hunk 2]
```diff
@@ -0,0 +1,584 @@
...
+                self.userCredentialModel.link(
+                  user.id, provider, authScheme, profile,
+                  {
+                    accessToken: accessToken,
+                    refreshToken: refreshToken
+                  }, options, done);
+              });
+            }
+          } else {
+            self.userIdentityModel.login(provider, authScheme, profile,
```
Comment (elropero) [inline middleware/sh-passport-configurator.js]: Another thing: For the `!link` case, we should look to see if there's already a userCredential with that FB ID. If there is, then we should NOT create an identityModel, and should instead log the user in with the LB user already associated with this FB ID. There are more permutations here, but this is one.

Comment (smkirkpatrick) [inline middleware/sh-passport-configurator.js]: Just to clarify, we still need to add the `userCredential` lookup for the `!link` case?

Comment (elropero) [inline middleware/sh-passport-configurator.js]: Yes.

**Pull request merged**